app/utils/helpers.ts

```diff
@@ -162,3 +162,48 @@ export function isMainActivity() {
         android: ShareModule?.getCurrentActivityName() === 'MainActivity',
     });
 }
+
+/**
+ * Extracts the domain name or IP address from a serverUrl.
+ *
+ * @param {string} input - The serverUrl, potentially with a port and protocol.
+ * @returns {string} The extracted domain name or IP address,
+ * or empty if no match is found.
+ *
+ * // Output:
+ * // Input: https://example-something.com => Result: example-something
+ * // Input: http://192.168.1.1 => Result: 192.168.1.1
+ * // Input: https://127.0.0.1:3000 => Result: 127.0.0.1
+ * // Input: https://subdomain.example.com/api => subdomain.example.com/api
+ * // Input: http://localhost:8080/app/v1 => localhost:8080/app/v1
+ */
+export function extractDomain(input: string) {
```

Comment:
what is the reason behind this function?

Reply:
is for creating the unique key to store in the server so we can identify the acknowledged per server, so i.e for http://10.0.0.2:8065 server, the key will be stored in the DB as: `${GLOBAL_IDENTIFIERS.PUSH_DISABLED_ACK}${extractCleanDomain(serverUrl)}`
<img width="1077" alt="Screenshot 2023-08-16 at 14 39 49" src="https://github.com/mattermost/mattermost-mobile/assets/10082627/b69c4027-a2da-40e7-9aa6-e1cc599e7aff">

Reply:
So two servers, same domain different port wont really work

Reply:
yes, you are right, that case would apply the other server acknowledged value. We can add the port to cover this scenario 👍 .